cloud9-toolkit-install.sh

```diff
@@ -0,0 +1,112 @@
+# Example:
+#     https://github.com/aws/aws-toolkit-vscode/releases/download/v1.24.0/aws-toolkit-vscode-1.24.0.vsix
+TOOLKIT_LATEST_ARTIFACT_URL="https://github.com/aws/aws-toolkit-vscode/releases/download/v${TOOLKIT_LATEST_VERSION}/aws-toolkit-vscode-${TOOLKIT_LATEST_VERSION}.vsix"
+# URL or local filepath pointing to toolkit VSIX or ZIP (containing a VSIX).
+TOOLKIT_FILE=${1:-}
+TOOLKIT_INSTALL_DIR="$HOME/environment/.c9/extensions/aws-toolkit-vscode"
+SCRIPT_WORKDIR="$HOME/environment/toolkit"
+
+_log() {
+    >&2 echo "$@"
+}
+
+# Runs whatever is passed.
+#
+# On failure:
+#   - prints the command output
+#   - exits the script
+_run() {
+    local out
+    if ! out="$(2>&1 $@)" ; then
+        _log "Command failed (output below): '${@}'"
+        echo "$out" | sed 's/^/    /'
+        _log "Command failed (output above): '${@}'"
+        exit 1
+    fi
+}
+
+_main() {
+(
+    if test -f "$TOOLKIT_FILE" ; then
+        # Ensure full path (before `cd` below).
+        TOOLKIT_FILE="$(readlink -f $TOOLKIT_FILE)"
+    fi
+
+    echo "Script will DELETE these directories:"
+    echo "    ${TOOLKIT_INSTALL_DIR}"
+    echo "    ${SCRIPT_WORKDIR}"
+    read -n1 -r -p "ENTER to continue, CTRL-c to cancel"
+    rm -rf "$TOOLKIT_INSTALL_DIR"
+    rm -rf "$SCRIPT_WORKDIR"
+
+    cd "$HOME/environment"
+    mkdir "${SCRIPT_WORKDIR}"
+    cd "${SCRIPT_WORKDIR}"
+
+    # Set default URL if no argument was provided.
+    if test -z "$TOOLKIT_FILE" ; then
+        _log "Latest release:"
+        _log "    URL    : $TOOLKIT_LATEST_RELEASE_URL"
+        _log "    version: $TOOLKIT_LATEST_VERSION"
+        _log "    VSIX   : $TOOLKIT_LATEST_ARTIFACT_URL"
+        TOOLKIT_FILE="$TOOLKIT_LATEST_ARTIFACT_URL"
+    fi
+
+    TOOLKIT_FILE_EXTENSION="${TOOLKIT_FILE: -4}"
+    if test -f "$TOOLKIT_FILE" ; then
+        _log "Local file (not URL): ${TOOLKIT_FILE}"
+        if [ "$TOOLKIT_FILE_EXTENSION" = ".zip" ] || [ "$TOOLKIT_FILE_EXTENSION" = ".ZIP" ] ; then
+            _log 'File is a .zip file'
+            _run unzip "$TOOLKIT_FILE"
+            _run unzip *.vsix
+        else
+            _log 'File is not .zip file, assuming .vsix'
+            _run unzip "$TOOLKIT_FILE"
+        fi
+    else
+        _log "File not found, treating as URL: ${TOOLKIT_FILE}"
+        _log 'Deleting toolkit.zip'
+        rm -rf toolkit.zip
+        _log 'Downloading...'
+        curl -o toolkit.zip -L "$TOOLKIT_FILE"
+        if [ "$TOOLKIT_FILE_EXTENSION" = ".zip" ] || [ "$TOOLKIT_FILE_EXTENSION" = ".ZIP" ] ; then
+            _log 'File is a .zip file'
+            _run unzip toolkit.zip
+            _run unzip *.vsix
+        else
+            _log 'File is not .zip file, assuming .vsix'
+            _run unzip toolkit.zip
+        fi
+    fi
+
+    mv extension "$TOOLKIT_INSTALL_DIR"
+    _log "Toolkit installed to: $TOOLKIT_INSTALL_DIR"
+    _log "Refresh Cloud9 to load it"
+)
+}
+
+_main
+
+# vim: et sw=4 ts=4
```

Comment:
remove? This file should get formatted by prettier as well?

Reply:
I found https://github.com/prettier/prettier/issues/4555 , seems like prettier doesn't support shell. I wouldn't trust it anyway, parsing bash/csh/zsh/sh is extremely non-trivial.

Reply:
- Added `prettier-plugin-sh` in https://github.com/aws/aws-toolkit-vscode/pull/1673/commits/8e37afa7f86b07d38d087fc320414e5895d45e6c 
- see https://github.com/aws/aws-toolkit-vscode/pull/1673/commits/bf15204443a0dfb13255a30b8a3035815d1eed68 for the result (mostly whitespace).